src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ApplicationPropertyPage.xaml

```diff
@@ -176,6 +179,24 @@
     </DynamicEnumProperty.Metadata>
   </DynamicEnumProperty>
 
+  <!-- TODO: Shortlink for help URL -->
+  <BoolProperty Name="UseApplicationFramework"
+                DisplayName="Application Framework"
+                Description="Enable the Visual Basic Application Framework for this project."
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/application-page-project-designer-visual-basic#enable-application-framework"
+                Category="General">
+    <BoolProperty.Metadata>
+      <NameValuePair Name="DependsOn" Value="Application::OutputType" />
+      <NameValuePair Name="VisibilityCondition">
+        <NameValuePair.Value>
+          (and
+            (has-evaluated-value "Application" "OutputType" "WinExe")
+            (not (has-csharp-lang-version-or-greater "1.0")))
```

Comment:
So, this is fun. We have logic to check the version of the C# language used. So, to make this for 'VB', I made it show up for when the C# language version is **NOT** greater than 1.0. This is just for development purposes. We'll need to add a real flag, specifically for VB.

Reply:
https://dev.azure.com/devdiv/DevDiv/_git/CPS/pullrequest/381213 adds `has-vb-lang-version-or-greater` which should help here.

Reply:
@drewnoakes Thank you! <3

Reply:
While talking with Drew, he had a good point: we should consider adding `has-vb-language` or similar in cases where we're only concerned with the language (not the version).

> So, this is fun. We have logic to check the version of the C# language used. So, to make this for 'VB', I made it show up for when the C# language version is **NOT** greater than 1.0. This is just for development purposes. We'll need to add a real flag, specifically for VB.

As a side note, the project properties UI is also enabled for F# so showing a property by `not (has-csharp-language)` would result in it being shown in VB and F#. More to the point in having a language-specific visibility condition. 😄

Reply:
Or:
- `has-language "VB"`
- `has-language "CSharp"`
- `has-lang-version "VB" "16.0" "<"`
- `has-lang-version "CSharp" "7.0" ">"`
- `has-lang-version "CSharp" "6.0" "="`

Just make every variant (something which has multiple potential values) a parameter. This allows selection of language, version, and scope (<, >, =). You could add (<=, >=) if needed.

Reply:
I've added functions `is-csharp` and `is-vb` in that PR as well.

The distinction between `is-` and `has-` comes down to configurations. You can configure different language versions in different configurations (for example) but you can't realistically configure different languages in different configurations.

We also shied away from requiring the user to pass variables such as "VB" or "CSharp" to make it harder to pass the wrong thing. For example, the above would not work. The `Language` property has value `"C#"` not `"CSharp"`.

Reply:
Note also that `(not (has-csharp-lang-version-or-greater "1.0"))` would match an F# project.